internal/resources/aws/lambda_function.go

```diff
@@ -86,6 +91,16 @@ func (a *LambdaFunction) BuildResource() *schema.Resource {
 	},
 	)
 
+	if a.MonthlyRequests != nil {
+		monthlyRequests = decimalPtr(decimal.NewFromInt(*a.MonthlyRequests))
+		gbSeconds = decimalPtr(calculateGBSeconds(memorySize, averageRequestDuration, *monthlyRequests))
+		storageGBSeconds = decimalPtr(calculateStorageGBSeconds(storageSize, *gbSeconds))
+
+		costComponents = append(costComponents, a.storageCostComponent(storageGBSeconds, storageType))
```

Comment:
I think we can move this logic down to be in the same block that we add the duration cost component, since we're already calculating `monthlyRequests` and `gbSeconds` there.